src/Resolvers/Cached.php

```diff
@@ -107,7 +113,7 @@ private function unwrapResults(array $results) : DNSRecordCollection
         $records = $results['recordCollection'];
         foreach ($records as $key => $record) {
             $records[$key] = $record
-                ->setTTL($record->getTTL() - ($this->getTimeStamp() - (int)$results['timestamp']));
+                ->setTTL(max($record->getTTL() - ($this->getTimeStamp() - (int)$results['timestamp']), 0));
```

Comment:
will we ever run into the default of 0? wouldn't `0` means that we cache forever?

Reply:
It's fine if the TTL is `0` here, so we can show what the api returned us for real, but the change to ignore `0` is here: https://github.com/remotelyliving/php-dns/pull/19/files#diff-148ec06d4b90abfec815d4126b81cd42R98